spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarProducerFactory.java

```diff
@@ -57,23 +57,43 @@ public DefaultPulsarProducerFactory(PulsarClient pulsarClient, Map<String, Objec
 
 	@Override
 	public Producer<T> createProducer(String topic, Schema<T> schema) throws PulsarClientException {
-		return createProducer(topic, schema, null, null);
+		return doCreateProducer(topic, schema, null, null, null);
 	}
 
 	@Override
 	public Producer<T> createProducer(String topic, Schema<T> schema, MessageRouter messageRouter)
 			throws PulsarClientException {
-		return createProducer(topic, schema, messageRouter, null);
+		return doCreateProducer(topic, schema, messageRouter, null, null);
 	}
 
 	@Override
 	public Producer<T> createProducer(String topic, Schema<T> schema, MessageRouter messageRouter,
 			List<ProducerInterceptor> producerInterceptors) throws PulsarClientException {
-		return doCreateProducer(topic, schema, messageRouter, producerInterceptors);
+		return doCreateProducer(topic, schema, messageRouter, producerInterceptors, null);
 	}
 
+	@Override
+	public Producer<T> createProducer(String topic, Schema<T> schema, MessageRouter messageRouter,
+			List<ProducerInterceptor> producerInterceptors,
+			List<ProducerBuilderCustomizer<T>> producerBuilderCustomizers) throws PulsarClientException {
+		return doCreateProducer(topic, schema, messageRouter, producerInterceptors, producerBuilderCustomizers);
+	}
+
+	/**
+	 * Create the actual producer.
+	 * @param topic the topic the producer will send messages to or {@code null} to use
+	 * the default topic
+	 * @param schema the schema of the messages to be sent
+	 * @param messageRouter the optional message router to use
+	 * @param producerInterceptors the optional producer interceptors to use
+	 * @param producerBuilderCustomizers the optional list of customizers to apply to the
+	 * producer builder
+	 * @return the created producer
+	 * @throws PulsarClientException if any error occurs
+	 */
 	protected Producer<T> doCreateProducer(String topic, Schema<T> schema, MessageRouter messageRouter,
```

Comment:
Creating this default `doCreateProducer` lets all of the other APIs call this directly and avoids chaining on one another to reduce blast radius on changes to the other APIs later down the road.